planner/core/testdata/derive_topn_from_window_in.json

```diff
@@ -0,0 +1,20 @@
+[
+  {
+    "name": "TestPushDerivedTopN",
+    "cases":[
+      "select  row_number() over (partition by a) from t -- pattern missing filter on row number",
+      "select * from (select rank() over () as rank_order from t) where rank_order <= 3 -- rank not supported in pattern yet",
+      "select * from (select row_number() over (partition by a) as rownumber from t) where rownumber <= 3 -- pattern is applicable but partition by is not supported yet",
+      "select * from (select row_number() over () as rownumber1, row_number() over (partition by a) as rownumber2 from t) where rownumber1 <= 3 -- pattern not applicable with multiple window functions",
+      "select * from (select b, row_number() over () as rownumber from t) where rownumber <= 3 and b > 5 -- pattern is not applicable with complex filter on top of window",
+      "select * from (select b, row_number() over () as rownumber from t) where rownumber > 3  -- pattern is not applicable with filter is not < or <=",
+      "select * from (select a,b, row_number() over () as rownumber from t) where  a > b  -- pattern is not applicable with filter is not < or <=",
+      "select * from (select a,b, row_number() over () as rownumber from t) where  a <= 3 -- pattern is not applicable with filter is not on row number",
+      "select * from (select a,b, row_number() over () as rownumber from t) where 3 >= rownumber  -- pattern is not applicable with filter is not < or <=",
+      "select * from (select row_number() over () as rownumber from t) where rownumber <= 3 -- pattern is applicable",
+      "select * from (select row_number() over () as rownumber from t) where rownumber < 3 -- pattern is applicable",
```

Comment:
Could you please add some cases with non-int constant values here like `rownumber < 3.0` and `rownumber < '3.0'`?

Reply:
Good point. Will do.